src/groupByKey.ts

```diff
@@ -1,7 +1,18 @@
+/* eslint-disable no-restricted-syntax */
 type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items, key): GroupsMap<T> {
```

Comment:
you can even strict type function parameters by using `(items: T[], key: keyof T): GroupsMap<T>`